### PR TITLE
Pthook for pythia8

### PR DESCRIPTION
**Pull request by alberto-sanchez**
This use the pT veto hook, which will allow to filter on quark presence before hadronization

### CODE REVIEW

**Comment by perrozzi** [comment]
please test

**Comment by perrozzi** [comment]
+1

File: GeneratorInterface/Pythia8Interface/src/PTFilterHook.cc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+//--------------------------------------------------------------------------
+bool PTFilterHook::checkVetoPT( int iPos, const Pythia8::Event& event) {
+
+  if (!filter_ || iPos >3 ) return false;
+  bool foundQuark = false;
+  
+  //look for quark
+  for (int i = 0; i < event.size(); ++i) {
+    if ( (abs(event[i].id()) == quark_ ) && ( abs(event[i].y()) <= quarkY_ ) ) {
+       double pT = sqrt(pow(event[i].px(),2) + pow(event[i].py(),2));
```
Comment (davidlange6) [inline GeneratorInterface/Pythia8Interface/src/PTFilterHook.cc]: @alberto-sanchez - use pT() function of event[i]?

Comment (alberto-sanchez) [inline GeneratorInterface/Pythia8Interface/src/PTFilterHook.cc]: @davidlange6:  Yes!. Sorry for that. I committed the wrong version of the code. I have fixed that. Thanks for catching this.

**Comment by perrozzi** [comment]
+1

**Comment by davidlange6** [comment]
@alberto-sanchez - please make the same change elsewhere in the PR where the code is repeated.

**Comment by alberto-sanchez** [comment]
@davidlange6 no more change are required, pT() is used just in that place.

**Comment by davidlange6** [comment]
+1
whoops -yes.

**Pull request merged**